### ISSUE TITLE
Feature Todo list: Montage Review

### ISSUE THREAD

**Issue by connortechnology** [ZoneMinder/zoneminder#4036]
Montage Review is fairly broken in it's design.  It currently constantly grab images one at a time and writes them to a canvas.  It seems that this cannot handle an mjpeg stream.  

The reason for the canvas is to be able to write text and the +/- buttons onto it.  I think we should replace it with the new layout from montage.  

OR

We copy over the history timeline logic to montage and scrap review.

**Comment by IgorA100** [comment]
>  canvas is  to be able to write text

I don’t quite understand what kind of text you are writing about?

**Comment by connortechnology** [comment]
If it can't load event data it writes "No data"over the image.  Really not needed. Can be done other ways.

**Comment by IgorA100** [comment]
Ok.

**Comment by IgorA100** [comment]
Is Live mode required? 
In essence, this is a duplication of the Montage page, but with FPS = 1.
FPS = 1-5 - this would be very good for a narrow network channel on both Montage and Montage Review

**Comment by connortechnology** [comment]
No it is not.  I would really like to merge these two views.  At the moment montage is much better.

I will be committing a branch tomorrow that adds an fps limiter to montage.

**Comment by IgorA100** [comment]
Why not make a single page for live viewing and for viewing recordings?
For example, you can make such a page based on Montage page.

**Comment by IgorA100** [comment]
@connortechnology 
I can't figure out where in PHP the "ZM_OPT_CONTROL" constant is defined?
I can't access it when working via AJAX.

**Comment by IgorA100** [comment]
@connortechnology 
What is this button for and does it work?
![5](https://github.com/ZoneMinder/zoneminder/assets/5006170/cce219e2-ba6a-48b1-a9d1-240a38ad489a)

When you press it, the camera control panel appears, but there is no image from the camera.
![6](https://github.com/ZoneMinder/zoneminder/assets/5006170/8ee3ac0c-7a6c-4391-8ea5-db40ad45ce31)

**Comment by connortechnology** [comment]
I think maybe you deleted the question about ZM_OPT_CONTROL.  For js land, all is provided by web/skins/classic/js/skin.js.php.  All but a few config variables are exposed there.

For php it is in web/includes/config.php.in (which becomes config.php).

**Comment by IgorA100** [comment]
> It is a leftover from the past.
...... perhaps it could just go away.

Got it, I’ll leave it “as is” for now, I won’t delete it. I thought that maybe something was broken on me. I also don’t see the point in this page if the camera can be controlled on the Watch page.

**Comment by connortechnology** [comment]
The only point would be to do so while watching multiple feeds.  Which is a valid use case, maybe you have multiple cameras from different angles watch a scene...

**Comment by IgorA100** [comment]
> The only point

Yes, it’s possible, but now I definitely won’t do it, because... a lot of work on partially converting the Montage page to Ajax and integrating it with Montage Review.

**Comment by IgorA100** [comment]
> I think maybe you deleted the question about ZM_OPT_CONTROL.

Yes, I deleted the message because... I have access to the ZM_OPT_CONTROL constant in my PHP code, but I still don’t understand where it is defined.

> For php it is in web/includes/config.php.in (which becomes config.php).

The fact of the matter is that there is no ZM_OPT_CONTROL constant in config.php
It probably comes from SQL somewhere and must be defined in PHP somewhere.

**Comment by connortechnology** [comment]
It is in the database.
config.php loads /etc/zm.conf to find out where the database is. Then loads the contents of the Config table.
